app/models/user.rb

```diff
@@ -116,10 +116,10 @@ def dummy_password_for_external_auth
   def change_password(oldpwd, newpwd)
     auth = self.class.authenticator(userid)
     raise MiqException::MiqEVMLoginError, "password change not allowed when authentication mode is #{auth.class.proper_name}" unless auth.uses_stored_password?
-    raise MiqException::MiqEVMLoginError, "old password does not match current password" unless User.authenticate(userid, oldpwd)
-
-    self.password = newpwd
-    self.save!
+    if User.authenticate(userid, oldpwd)
```

Comment:
To be honest, I did not look at what exactly the user sees at this point, since this should do essentially exactly what it currently does (previous code erroneously assumes the exception here is thrown `unless User.authenticate(..)`, which isn't true). 

I'll look though.